### DISCUSSION TITLE
Feature for removing all tables that are not used by current models

### DISCUSSION THREAD

**Discussion by JavierLopezT** [dbt-labs/dbt-core#7702] [Ideas]
There are some cases in which all the tables we have in our database are fully managed by dbt, either in the form of seeds, sources or tables/views. 

Sometimes I make changes that leave old tables/views behind (like changing the name of a seed) and it would be nice to have a command like:

`dbt sync`

That removes all the old tables.

There have been custom solutions to that in threads like:
https://discourse.getdbt.com/t/clean-your-warehouse-of-old-and-deprecated-models/1547/10
https://discourse.getdbt.com/t/faq-cleaning-up-removed-models-from-your-production-schema/113/13

But I think it would be nice to have it as a core feature. 

What do you think?

**Comment by jtcohen6** [comment]
@JavierLopezT Check out this proposal & in-progress work:
- https://github.com/dbt-labs/dbt-core/issues/4957
- https://github.com/dbt-labs/dbt-core/pull/5392

**Comment by alison985** [comment]
This at least semi-relates to https://github.com/dbt-labs/dbt-core/issues/8194 FYI @damian3031
